platforms/jvm/platform-jvm/src/main/java/org/gradle/platform/Architecture.java

```diff
@@ -36,7 +39,66 @@ public enum Architecture {
     X86_64,
 
     /**
-     * 64-bit reduced instruction set computer (RISC) architectures, including "aarch64", "arm64".
+     * 64-bit reduced instruction set computer (RISC) ARM architectures, including "aarch64", "arm64".
+     */
+    AARCH64,
+
+    /**
+     * 64-bit IBM PowerPC (big-endian)
+     * @since 8.6
+     */
+    PPC64,
+
+    /**
+     * 64-bit IBM PowerPC (little-endian)
+     * @since 8.6
+     */
+    PPC64LE,
+
+    /**
+     * 64-bit IBM Z architectures (historically called "s390x" on Linux)
+     * @since 8.6
+     */
+    S390X,
+
+    /**
+     * 64-bit Sun/Oracle SPARC V9
+     * @since 8.6
+     */
+    SPARC_V9;
+
+    /**
+     * Get the architecture of the current machine
+     * @since 8.6
+     */
+    @Nonnull
+    public static Architecture current() {
+        return forName(System.getProperty("os.arch", "none"));
+    }
+
+    /**
+     * Determine the architecture from the value of the "os.arch" system property
+     * @since 8.6
```

Comment:
This javadoc incorrectly suggests it uses the system property in the method, but it actually just wants a value that could have come from it:
```suggestion
     * Determine the architecture from a specific string value.
     *
     * @param arch a value from the {@code "os.arch"} system property
     * @since 8.6
```